cli/src/main/java/org/openjdk/skara/cli/GitSync.java

```diff
@@ -0,0 +1,164 @@
+        pb.inheritIO();
+        return pb.start().waitFor();
+    }
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+        var flags = List.of(
+            Option.shortcut("")
+                  .fullname("branches")
+                  .describe("BRANCHES")
+                  .helptext("Comma separated list of branches to sync")
+                  .optional(),
+            Switch.shortcut("")
+                  .fullname("pull")
+                  .helptext("Pull current branch from origin after successful sync")
+                  .optional(),
+            Switch.shortcut("")
+                  .fullname("fetch")
+                  .helptext("Fetch current branch from origin after successful sync")
+                  .optional(),
+            Switch.shortcut("m")
+                  .fullname("mercurial")
+                  .helptext("Force use of mercurial")
+                  .optional(),
+            Switch.shortcut("")
+                  .fullname("verbose")
+                  .helptext("Turn on verbose output")
+                  .optional(),
+            Switch.shortcut("")
+                  .fullname("debug")
+                  .helptext("Turn on debugging output")
+                  .optional(),
+            Switch.shortcut("v")
+                  .fullname("version")
+                  .helptext("Print the version of this tool")
+                  .optional()
+        );
+
+        var inputs = List.of(
+            Input.position(0)
+                 .describe("REMOTE")
+                 .singular()
+                 .optional()
+        );
+
+        var parser = new ArgumentParser("git sync", flags, inputs);
+        var arguments = parser.parse(args);
+
+        if (arguments.contains("version")) {
+            System.out.println("git-sync version: " + Version.fromManifest().orElse("unknown"));
+            System.exit(0);
+        }
+
+        if (arguments.contains("verbose") || arguments.contains("debug")) {
+            var level = arguments.contains("debug") ? Level.FINER : Level.FINE;
+            Logging.setup(level);
+        }
+
+        var cwd = Paths.get("").toAbsolutePath();
+        var repo = Repository.get(cwd).orElseThrow(() ->
+                die("error: no repository found at " + cwd.toString())
+        );
+
+        var remotes = repo.remotes();
+
+        String upstream = null;
+        if (arguments.at(0).isPresent()) {
+            upstream = arguments.at(0).asString();
+        } else {
+            var lines = repo.config("sync.remote");
+            if (lines.size() == 1 && remotes.contains(lines.get(0))) {
+                upstream = lines.get(0);
+            } else {
+                die("No remote provided to sync with");
+            }
+        }
+
+        var upstreamPullPath = remotes.contains(upstream) ?
+            Remote.toURI(repo.pullPath(upstream)) : URI.create(upstream);
+        var origin = "origin";
```

Comment:
I think the fork remote name should be a CLI option as well, maybe in the gitconfig as well (e.g. I have named it `fork` for some of my local repos)

Reply:
Sounds good!